### PR TITLE
Provide chunk size suggestion for BnP jobs with chunk overflow except…

### PR DESCRIPTION
**Pull request by mattwisein**
…ions and fix num chunks algorith to round up

### CODE REVIEW

File: contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java [hunk 1]
```diff
@@ -289,15 +280,39 @@ public void build() {
...
-            RunningJob job = JobClient.runJob(conf);
 
-            // Once the job has completed log the counter
-            Counters counters = job.getCounters();
+            // The snipped below copied and adapted from: JobClient.runJob(conf);
+            // We have more control in the error handling this way.
+
+            JobClient jc = new JobClient(conf);
+            RunningJob runningJob = jc.submitJob(conf);
+            Counters counters;
```
Comment (mattwisein) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java]: Counters now only defined in outer scope

File: contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java [hunk 2]
```diff
@@ -289,15 +280,39 @@ public void build() {
...
+
+            try {
+                if (!jc.monitorAndPrintJob(conf, runningJob)) {
+                    counters = runningJob.getCounters();
```
Comment (mattwisein) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java]: but counters are set after job is done.

File: contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java [hunk 3]
```diff
@@ -289,15 +280,39 @@ public void build() {
...
+                    } else {
+                        logger.error("Job Failed: " + runningJob.getFailureInfo());
+                    }
+                    throw new VoldemortException("BnP's MapReduce job failed.");
+                }
+            } catch (InterruptedException ie) {
+                Thread.currentThread().interrupt();
+            }
 
+            counters = runningJob.getCounters();
```
Comment (mattwisein) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java]: And set again outside of the if block.

**Comment by FelixGV** [comment]
This change LGTM.

Can you squash both commits into one? We usually try to have a short first line so that GitHub doesn't clip it. Then we can add lengthier details on another paragraph below. Please commit the whole thing under your name, you deserve the brownie points for driving the implementation and stabilization to completion ;)

**Comment by mattwisein** [comment]
Thanks @FelixGV .  I've updated the author, and it's only one commit.  I'll plan to merge this to master by EOD unless someone complains.

File: contrib/hadoop-store-builder/src/java/voldemort/store/readonly/disk/HadoopStoreWriter.java [hunk 4]
```diff
@@ -55,6 +55,27 @@
...
+     * For each chunk in a partition, Index files contains the integer offset into
+     * the corresponding data file. So it is capped at {@link Integer#MAX_VALUE}.
+     */
+    public static final long MAX_CHUNK_SIZE = Integer.MAX_VALUE;
+
+    /**
+     * The default chunk size tries to leave an error margin so that compressed
+     * data size on hadoop or skewed data sets don't cause chunk overflows
+     */
+    public static final long DEFAULT_CHUNK_SIZE = MAX_CHUNK_SIZE / 2;
```
Comment (gaojieliu) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/disk/HadoopStoreWriter.java]: Is this 'DEFAULT_CHUNK_SIZE' being used in VoldemortBuildAndPushJob.java?
I saw it is still using '1024 * 1024 * 1024'.
https://github.com/voldemort/voldemort/blob/master/contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java#L744

Comment (FelixGV) [inline contrib/hadoop-store-builder/src/java/voldemort/store/readonly/disk/HadoopStoreWriter.java]: Good catch @gaojieliu.

We should use the new constant in the BnP job!

**Comment by mattwisein** [comment]
This is merged.

**Pull request closed without merging**